File: samples/java/src/main/java/com/example/DatabaseConfig.java
package com.example;

import static com.example.Constants.DB_HOST_KEY;
import static com.example.Constants.DB_PORT_KEY;

/**
 * Database connection configuration resolved from environment variables.
 */
public class DatabaseConfig {

    // Feature: System.getenv(CONST_REF) — cross-file constant reference
    /** Hostname of the primary database server. */
    public final String host = System.getenv(DB_HOST_KEY);

    // Feature: System.getenv().getOrDefault(CONST_REF, "default") — const ref in getOrDefault
    /** Port the database listens on. */
    public final String port = System.getenv().getOrDefault(DB_PORT_KEY, "5432");

    // Feature: System.getenv("LITERAL") — literal key, required (no default available)
    /** Name of the database schema to connect to. */
    public final String name = System.getenv("DATABASE_NAME");

    // Feature: System.getenv().getOrDefault("LITERAL", "default") — literal key with fallback
    /** Maximum number of pooled database connections. */
    public final String maxConnections = System.getenv().getOrDefault("DB_MAX_CONNECTIONS", "10");
}

File: samples/basic/src/main/kotlin/DatabaseConfig.kt
package com.example

// Feature: System.getenv(CONST_REF) — cross-file constant reference + Elvis default
/** Hostname of the primary database server. */
val dbHost = System.getenv(DB_HOST_KEY) ?: "localhost"

// Feature: System.getenv().getOrDefault(CONST_REF, "default") — cross-file const in getOrDefault
/** Port the database listens on. */
val dbPort = System.getenv().getOrDefault(DB_PORT_KEY, "5432")

// Feature: System.getenv("LITERAL") + Elvis default
/** Maximum number of pooled database connections. */
val dbMaxConnections = System.getenv("DB_MAX_CONNECTIONS") ?: "10"

// Feature: System.getenv("LITERAL") + Elvis throw → required=true, default=null
/** Name of the database schema to connect to. */
val dbName = System.getenv("DATABASE_NAME") ?: throw IllegalStateException("DATABASE_NAME is required")
